app/code/Magento/CatalogCustomerGraphQl/Model/Resolver/PriceTiers.php

```diff
@@ -158,4 +158,45 @@ private function formatProductTierPrices(array $tierPrices, float $productPrice,
         }
         return $tiers;
     }
+
+    /**
+     * Select a lower price for each quantity
+     *
+     * @param ProductTierPriceInterface[] $tierPrices
+     *
+     * @return array
+     */
+    private function filterTierPrices(array $tierPrices): array
+    {
+        $qtyCache = [];
+        foreach ($tierPrices as $item => &$price) {
```

Comment:
We are passing the `&$price` variable by reference but it does not seem like we modify it somehow in the loop below. But I may miss something.

Reply:
**Fixed**

Reply:
@Usik2203 
From "clarity of intent" perspective, when one reads through the "resolve" function alone, it seems as though formatProductTierPrices function does only what its name says. In fact this function's comment seems to support that.
But, that is misleading.

So, I am wondering can we have a function named like filterAndFormatProductTierPrices, that invokes a foreach loop, and inside this loop, one below the other filterTierPrices and formatProductTierPrices are called?

Check for example, when an unfortunate situation like we have, calls for two different things to be done in the same method:
`League\Flysystem\Filesystem::readAndDelete()`
internally first calls
`$this->read()`
and later
`$this->delete()`

Reply:
@pmarjan **Done**